adversarial_text/data/data_utils.py

```diff
@@ -215,13 +215,16 @@ def build_lm_sequence(seq):
 
   Returns:
     SequenceWrapper with `seq` tokens copied over to output sequence tokens and
-    labels (offset by 1, i.e. predict next token) with weights set to 1.0.
+    labels (offset by 1, i.e. predict next token) with weights set to 1.0, except for <eos> token.
   """
   lm_seq = SequenceWrapper()
-  for i, timestep in enumerate(seq[:-1]):
-    lm_seq.add_timestep().set_token(timestep.token).set_label(
+  for i, timestep in enumerate(seq):
```

Comment:
Because The last 'eos' is necessary for calculating virtual adversarial loss(virtual adversarial loss uses lm_inputs).  
The classification model predict a label on 'eos'. please look at the figure 1 in the paper : https://arxiv.org/pdf/1605.07725.pdf. 
If there is no 'eos' token in the lm_inputs, the classification model cannot predict label correctly on the sequence.